### ISSUE TITLE
Extension not working, "command 'vscode-objectscript.explorer.refresh' not found"

### ISSUE THREAD

**Issue by programx** [intersystems-community/vscode-objectscript#586]
Issue Type: <b>Bug</b>

On installing the ObjectScript and/or the InterSystems Extension pack, the extension doesn't work. It simply displays the panel, which is empty. Additionally, clicking the "Refresh" icon just displays the message: "command 'vscode-objectscript.explorer.refresh' not found"

Extension version: 1.0.8
VS Code version: Code 1.54.3 (2b9aebd5354a3629c3aba0a5f5df49f43d6689f8, 2021-03-15T10:55:45.459Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-4590S CPU @ 3.00GHz (4 x 2993)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.91GB (4.79GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstry914:30276682
pythonvsdeb440:30248342
pythonvsded773:30248341
pythonvspyt875:30259475
dockersubset:30270708
pythontbcf:30265426
vscoresta800cf:30276762
vspre833cf:30267465
vsdfh931:30275552
vshan820cf:30276953

```

</details>

![image](https://user-images.githubusercontent.com/12547342/111421499-008ff800-86e5-11eb-9cf5-8e9fc0370e6e.png)

**Comment by programx** [comment]
I should say I have tried reinstalling the extensions and VS Code in various orders and trawled my disk to remove any trace of VS Code and its extensions.

**Comment by gjsjohnmurray** [comment]
Have you worked through this README section?

https://github.com/intersystems-community/vscode-objectscript#configure-a-connection

**Comment by programx** [comment]
Thanks @gjsjohnmurray . Yes, I have run through. I have/had configurations sections defined for the servers, etc.

Here's an excerpt, if it helps ...

`   
        

        "intersystems.servers": {
            "ourserver": {
            "webServer": {
                "scheme": "http",
                "host": "ourserver",
                "port": 57772
            },
            "description": "Our server"
        },
        "default~cache": {
            "webServer": {
                "scheme": "http",
                "host": "127.0.0.1",
                "port": 57772
            },
            "description": "Connection to local InterSystems Caché installed with default settings."
        },
        "default~ensemble": {
            "webServer": {
                "scheme": "http",
                "host": "127.0.0.1",
                "port": 57772
            },
            "description": "Connection to local InterSystems Ensemble installed with default settings."
        },
        "/default": "ourserver"
    }`

**Comment by gjsjohnmurray** [comment]
What does your `objectscript.conn` setting contain?

Do you see a status bar item contributed by this extension? What does it show?

**Comment by programx** [comment]
My `objectscript.conn` setting:

`"objectscript.conn": {
        "active": true,
        "host": "ourserver",
        "https": false,
        "ns": "OURNAMESPACE",
        "password": "password-here-only-doe-diagnostics",
        "port": 57772,
        "server": "ourserver",
        "username": "npw0422"
    },`

I see no discernible status bar text:
![image](https://user-images.githubusercontent.com/12547342/111433235-f1fe0c80-86f5-11eb-8b5b-62a54d1e2cbc.png)

**Comment by gjsjohnmurray** [comment]
@programx sorry for the slow response.

I can reproduce what you're seeing if I set the `server` property of `objectscript.conn` so that it names a server which doesn't exist in my `intersystems.servers` object.

The `objectscript.conn` you reported above has `"server: "ourserver"` which matches an entry in the `intersystems.servers` object that you previously reported.

Did that `objectscript.conn` come from your user-level settings.json?

Does the folder you are opening (which your screenshot indicates is named 'cache-in-vscode-test`) contain its own settings.json file in a .vscode subfolder? Of so, maybe that file (workspace-level settings) defines an `objectscript.conn` with a `server` property pointing to a non-existent `intersystems.servers` entry.

Either way it's worth you knowing that if your `objectscript.conn` specifies `server` then the properties `host`, `port`, `https`, `username` and `password` are all ignored. Instead, those values are fetched from the `intersystems.servers` entry the `server` property points to. I recommend you remove them from your setting to avoid confusion.

**Comment by taggloo** [comment]
Thanks very much John.

What you say makes sense. I have had to redact all my identifiable data in my examples for obvious reasons so am unsure what actually happened. But it looks like I was on to something.

Now, getting back to it, it all works! I changed nothing.

**Comment by programx** [comment]
Sorry, this was me ^^^ my password manager auto-filled my login fields and I didn't catch how.

**Comment by gjsjohnmurray** [comment]
@programx thanks for the update. FYI the fix referenced earlier in this issue is in 1.0.9 which we just released.
